pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -187,14 +190,29 @@ func (cl *Client) Run(stop <-chan struct{}) {
 	cl.SyncAll()
 	cl.initialSync.Store(true)
 	scope.Info("Pilot K8S CRD controller synced ", time.Since(t0))
+
+	cl.crdMetadataInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			crd, ok := obj.(*metav1.PartialObjectMetadata)
+			if !ok {
+				// Shouldn't happen
+				scope.Errorf("wrong type %T: %v", obj, obj)
+				return
+			}
+			handleCRDAdd(cl, crd.Name, stop)
```

Comment:
This is time costing, it is not a good practice to synchronously handle event

Reply:
I think the only place this may block is in the waitforcachesync part. I may be able to remove that - if so there is nothing doing any IO here so it should be fast enough?

if not maybe can move this into go routine?